### PR TITLE
User + friend protobuf

### PR DESCRIPTION
**Pull request by ming-00**
Will work on the backend if this is okay!

Btw I wanted to use PersonInfo in friend.proto but had trouble importing it successfully 😅

### CODE REVIEW

File: api-specs/friend.proto [hunk 1]
```diff
@@ -0,0 +1,16 @@
+syntax = "proto3";
+package friendsapi;
+option go_package = "/friendsapi"
+
+import "google/protobuf/empty.proto";
+
+service FriendService {
+	rpc Add (FriendRequest) returns (google.protobuf.Empty);
+	rpc Accept (FriendRequest) returns (google.protobuf.Empty);
+	rpc Block (FriendRequest) return (google.protobuf.Empty);
```
Comment (ianyong) [inline api-specs/friend.proto]:
```suggestion
	rpc Block (FriendRequest) returns (google.protobuf.Empty);
```

File: api-specs/user.proto [hunk 2]
```diff
@@ -0,0 +1,17 @@
+syntax = "proto3";
+package userapi;
+option go_package = "/userapi";
```
Comment (ianyong) [inline api-specs/user.proto]:
```suggestion
option go_package = "/usersapi";
```

For consistency.

File: api-specs/friend.proto [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+}
+
+message FriendRequest {
+	int64 senderId = 1;
+	int64 receiverId = 2;
+}
```
Comment (ianyong) [inline api-specs/friend.proto]:
```suggestion
}

```

Missing newline at EOF.

File: api-specs/user.proto [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+message getProfileRequest {
+	int64 userId = 1;
+}
+
+message Profile {
+	string name = 1;
+	string username = 2;
+	repeated Profile friends = 3;
+}
```
Comment (ianyong) [inline api-specs/user.proto]:
```suggestion
}

```

File: api-specs/user.proto [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+service userService {
+	rpc getProfile (getProfileRequest) returns (Profile);
+}
+
+message getProfileRequest {
+	int64 userId = 1;
+}
+
+message Profile {
```
Comment (ianyong) [inline api-specs/user.proto]: Should we make use of `ProfileRequest` and `ProfileResponse` as a convention?

**Review by ianyong — CHANGES_REQUESTED** [review]

File: api-specs/user.proto [hunk 6]
```diff
@@ -0,0 +1,17 @@
+syntax = "proto3";
+package userapi;
+option go_package = "/userapi";
+
+service userService {
```
Comment (sebastiantoh) [inline api-specs/user.proto]:
```suggestion
service UserService {
```

for consistency as well haha

File: api-specs/user.proto (hunk 5 above)
Comment (ming-00) [reply api-specs/user.proto]: okay!

File: api-specs/friend.proto [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+import "google/protobuf/empty.proto";
+
+service FriendService {
+	rpc Add (FriendRequest) returns (google.protobuf.Empty);
+	rpc Accept (FriendRequest) returns (google.protobuf.Empty);
+	rpc Block (FriendRequest) return (google.protobuf.Empty);
+}
+
+message FriendRequest {
```
Comment (okg00dbye) [inline api-specs/friend.proto:16]: do we need timestamp for date of request(?) to be able to display them by latest

File: api-specs/friend.proto (hunk 1 above)
Comment (okg00dbye) [inline api-specs/friend.proto]: possible to add GetPendingFriendRequests ?

File: api-specs/user.proto [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+	rpc getProfile (getProfileRequest) returns (Profile);
```
Comment (okg00dbye) [inline api-specs/user.proto]: thinking if we should separate get username/displayname and get friends as separate methods, possible usecase is user just made changes to their frens list and we need fetch list of frens again, wdyall think

File: api-specs/friend.proto (hunk 1 above)
Comment (okg00dbye) [inline api-specs/friend.proto]: sorry what does Block FriendRequest do

File: api-specs/friend.proto (hunk 7 above)
Comment (ming-00) [reply api-specs/friend.proto:16]: Good idea!

**Review by ianyong — APPROVED** [review]
LGTM!

File: api-specs/user.proto (hunk 8 above)
Comment (ming-00) [reply api-specs/user.proto]: oops missed this, I'll modify as you suggested!

File: api-specs/friend.proto (hunk 1 above)
Comment (ming-00) [reply api-specs/friend.proto]: It's our block a user, but not sure if we gonna have time to implement right :')

**Pull request merged**